src/sql/src/rbac.rs

```diff
@@ -977,6 +978,38 @@ fn generate_rbac_requirements(
                 ..Default::default()
             }
         }
+        Plan::AlterSchemaSwap(plan::AlterSchemaSwapPlan {
+            schema_a_spec,
+            schema_a_name: _,
+            schema_b_spec,
+            schema_b_name: _,
+            name_temp: _,
+        }) => {
+            let mut privileges = vec![];
+            if let ResolvedDatabaseSpecifier::Id(id_a) = schema_a_spec.0 {
+                privileges.push((
+                    SystemObjectId::Object(ObjectId::Database(id_a)),
+                    AclMode::CREATE,
```

Comment:
Do we need to care about DELETE or USAGE on the database or schemas? What does postgres do?

Reply:
Postgres doesn't have a swap notation, but for renames requires ownership of the items. [Snowflake](https://docs.snowflake.com/en/sql-reference/sql/alter-table) requires ownership of the object and create privileges in the owning schema/database, which I mimicked here.

FWIW [Joe advised me on the original rbac](https://github.com/MaterializeInc/materialize/pull/22365#discussion_r1357447994) perms when adding RENAME for schemas